Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/TestData/abs.materialcanvasnode.azasset

```diff
@@ -0,0 +1,57 @@
+{
+    "Type": "JsonSerialization",
+    "Version": 1,
+    "ClassName": "DynamicNodeConfig",
+    "ClassData": {
+        "category": "Math Functions",
+        "title": "abs",
+        "inputSlots": [
+            {
+                "name": "inValue",
+                "displayName": "Value",
+                "description": "Value",
+                "supportedDataTypes": [
+                    "float",
+                    "float2",
+                    "float3",
+                    "float4",
+                    "color"
+                ],
+                "defaultValue": {
+                    "$type": "Vector4",
+                    "Value": [
+                        0.0,
+                        0.0,
+                        0.0,
+                        0.0
+                    ]
+                },
+                "settings": {
+                    "instructions": [ "SLOTTYPE NODEID_SLOTNAME = SLOTVALUE;" ]
+                }
+            }
+        ],
+        "outputSlots": [
+            {
+                "name": "outValue",
+                "displayName": "Value",
+                "description": "Value",
+                "supportedDataTypes": [
+                    "float4"
```

Comment:
All the ops here accept a variable component-size input, but seem to return a `float4`. Shouldn't the output component size match the input component size?

Reply:
Yes, as mentioned above, currently using the type from “this” slot instead of the “source” slot providing the value. Ran into that playing around with some of the nodes where the function should have worked  but referenced the wrong type.